src/EnergyPlus/InputProcessing/IdfParser.cc

```diff
@@ -248,11 +238,11 @@ json IdfParser::parse_idf(std::string const &idf, size_t &index, bool &success,
                 auto found_index = idf.find_first_of('\n', beginning_of_line_index);
                 std::string line;
                 if (found_index != std::string::npos) {
-                    line = idf.substr(beginning_of_line_index, found_index - beginning_of_line_index);
+                    line = idf.substr(beginning_of_line_index, found_index - beginning_of_line_index - 1);
                 }
                 errors_.emplace_back(fmt::format("Line: {} Index: {}", cur_line_num, index_into_cur_line) + " - Error parsing \"" + obj_name +
                                      "\". Error in following line.");
-                errors_.emplace_back("~~~ " + line);
+                errors_.emplace_back(fmt::format("~~~ {}", line));
```

Comment:
This was the first attempt to fix the newline problem, decided to just leave it.